### PR TITLE
Output `labels.json`, package updates, Node.js bump

### LINKED ISSUE #19: Update Node.js to active LTS version
## What

Update this project to use the latest active Long Term Support (LTS) version of Node.js for development, testing, and deployment.

## Why

This project currently uses Node.js version 12, which reaches end of life at the end of March this year (2022-04-30).

## Who needs to work on this

Developer

## Who needs to review this

Developer, Tech lead

## Done when

- [x] Update `.nvmrc`
- [x]  Update `package-lock.json` to use lockfile v3

### PR DESCRIPTION
**Pull request by colinrotherham**
Bit of a general "spruce up" PR to bring things up to date again 😊 

Includes a variety of updates:

1. Add npm script `build` to [output `labels.json`](https://github.com/Financial-Times/github-label-sync#label-config-file)
2. Add npm script `lint` for [JavaScript Standard Style](https://standardjs.com)
3. Upgrade to [`github-label-sync@2.2.0`](https://github.com/Financial-Times/github-label-sync/releases/tag/v2.2.0)
4. Other package updates (including [`chalk@5`](https://github.com/chalk/chalk/releases/tag/v5.0.0) which is now ESM only)
5. GitHub Actions updates
6. Node.js LTS update
7. Dependabot 🤖 

Also added npm scripts for `preview` and `test` for consistency too

Closes https://github.com/alphagov/design-system-team-labels/issues/19 by tracking Node.js LTS updates

* https://github.com/alphagov/design-system-team-labels/issues/19

### CODE REVIEW

**Comment by colinrotherham** [comment]
Suggest we hold off this PR until we merge:

* https://github.com/alphagov/design-system-team-labels/pull/25

@lfdebrux and @kellylee-gds are also going to audit our labels and bring the config up to date

(We can rebase this branch after)

**Comment by lfdebrux** [comment]
@colinrotherham `config.js` is now up-to-date, this branch needs rebasing though.

**Comment by colinrotherham** [comment]
> @colinrotherham `config.js` is now up-to-date, this branch needs rebasing though.

Thanks. Rebased and ready for review

All feedback accepted

File: apply.sh
Comment (colinrotherham) [inline apply.sh:9]: We're now using the default **labels.json**

Newer releases of [`github-label-sync`](https://github.com/Financial-Times/github-label-sync#readme) have added support for [`*.yaml` config files](https://github.com/Financial-Times/github-label-sync/pull/47) (in addition to `*.json`) but the changes meant we lost `*.js` config support via `require()` resolution

**Comment by lfdebrux** [comment]
@colinrotherham why do we need to rename the `js` files to `mjs` to use ES modules?

File: build.mjs
Comment (lfdebrux) [inline build.mjs]: Why is this preferable to `path.join(__dirname, 'labels.json')`?

**Review by lfdebrux — APPROVED** [review]

File: build.mjs
Comment (colinrotherham) [reply build.mjs]: Sadly got removed in ESM


### No `__filename` or `__dirname`
https://nodejs.org/api/esm.html#no-__filename-or-__dirname

>`__filename` and `__dirname` use cases can be replicated via [`import.meta.url`](https://nodejs.org/api/esm.html#importmetaurl).

**Comment by colinrotherham** [comment]
> @colinrotherham why do we need to rename the `js` files to `mjs` to use ES modules?

@lfdebrux Ah usually I'd _opt-in_ to ESM with `*.mjs` so "assumed CommonJS" configs still work:

```console
.eslintrc.js
babel.config.js
gulpfile.js
jest.config.js
postcss.config.js
puppeteer.config.js
```

It's a lot harder to _opt-out_ from ESM with `*.cjs` due to package support of `*.js` only

We have none of these so if you'd prefer I can flip the big **package.json** switch?

```json
{
  "type": "module"
}
```

**Comment by lfdebrux** [comment]
> > @colinrotherham why do we need to rename the `js` files to `mjs` to use ES modules?
> 
> @lfdebrux Ah usually I'd _opt-in_ to ESM with `*.mjs` so "assumed CommonJS" configs still work:
> 
> ```
> .eslintrc.js
> babel.config.js
> gulpfile.js
> jest.config.js
> postcss.config.js
> puppeteer.config.js
> ```
> 
> It's a lot harder to _opt-out_ from ESM with `*.cjs` due to package support of `*.js` only
> 
> We have none of these so if you'd prefer I can flip the big **package.json** switch?
> 
> ```json
> {
>   "type": "module"
> }
> ```

No change needed, just asking out of curiosity.

File: build.mjs
Comment (lfdebrux) [reply build.mjs]: Ah, I didn't know that, thanks.

**Comment by colinrotherham** [comment]
> No change needed, just asking out of curiosity.

Thanks @lfdebrux, let's give it a try and switch to "pure ESM"

This GitHub gist from the from the [`chalk@5` release notes](https://github.com/chalk/chalk/releases/tag/v5.0.0) helps explain some issues:
https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c

Just note that future packages might expect `*.js` to be CommonJS (not ESM)

**Comment by colinrotherham** [comment]
Merging now, let's see how the update to [`github-label-sync@2.2.0`](https://github.com/Financial-Times/github-label-sync/releases/tag/v2.2.0) gets on

**Pull request merged**
